### PR TITLE
Adaptions to render bfme map objects

### PR DESCRIPTION
**Pull request by Tarcontar**

### CODE REVIEW

**Comment by tgjones** [comment]
You probably want to rebase this sooner rather than later, to resolve the conflicts in ParticleSystem (those changes are no longer needed).

**Comment by Tarcontar** [comment]
Already rebased a few hours ago

**Comment by tgjones** [comment]
Ah, okay. And then please just revert the “readonly” changes in ParticleSystems.ParticleSystem.

File: src/OpenSage.Game/Assets/Shaders/FixedFunction.frag [hunk 1]
```diff
@@ -237,6 +241,20 @@ vec4 SampleTexture(
             // TODO: Haven't seen any non-zero values to test this with yet.
             break;
         }
+
+        case TEXTURE_MAPPING_WS_ENVIRONMENT:
+        {
+            //TODO: this is just a copy of ENVIRONMENT
```
Comment (tgjones) [inline src/OpenSage.Game/Assets/Shaders/FixedFunction.frag]: Please add a space after the //, in all comments.

File: src/OpenSage.Game/Content/MapLoader.cs [hunk 2]
```diff
@@ -298,7 +298,12 @@ private static GameObject CreateGameObject(MapObject mapObject, Team[] teams, Co
 
             if (mapObject.Properties.TryGetValue("originalOwner", out var teamName))
             {
-                var team = teams.FirstOrDefault(t => t.Name == (string) teamName.Value);
+                var name = (string)teamName.Value;
```
Comment (tgjones) [inline src/OpenSage.Game/Content/MapLoader.cs]: Please add a space after casts.

File: src/OpenSage.Game/Graphics/ParticleSystems/ParticleSystem.cs [hunk 3]
```diff
@@ -152,6 +152,7 @@ private static BlendStateDescription GetBlendState(ParticleSystemShader shader)
             switch (shader)
             {
                 case ParticleSystemShader.Alpha:
+                case ParticleSystemShader.AlphaTest:
```
Comment (tgjones) [inline src/OpenSage.Game/Graphics/ParticleSystems/ParticleSystem.cs]: Please add a TODO here.

File: src/OpenSage.Game/Logic/Object/Draw/W3dModelDraw.cs [hunk 4]
```diff
@@ -191,9 +191,9 @@ private W3dModelDrawConditionState CreateModelDrawConditionStateInstance(ModelCo
                     }
 
                     particleSystems.Add(new ParticleSystem(
-                        _contentManager,
-                        particleSystemTemplate,
-                        () => ref modelInstance.AbsoluteBoneTransforms[bone.Index]));
+                    _contentManager,
```
Comment (tgjones) [inline src/OpenSage.Game/Logic/Object/Draw/W3dModelDraw.cs]: Any reason for this whitespace change?

**Pull request merged**